File: dev-docs/snippets/workflows/confirm-apply.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Confirm a claimed apply once the output has been written to the document.
// Transitions the run from `apply_claimed` to `completed`.
export async function confirmApply(
  client: JsBaoClient,
  workflowKey: string,
  runKey: string,
  contextDocId: string,
) {
  // #region example
  const result = await client.workflows.confirmApply({
    workflowKey,
    runKey,
    contextDocId,
  });
  // result.confirmed
  // #endregion example
  return result;
}

File: dev-docs/snippets/workflows/confirm-apply.swift
import JsBaoClient

// Confirm a claimed apply once the output has been written to the document.
// Conditional on the same connection that called `claimApply`.
func confirmApply(
  client: JsBaoClient,
  workflowKey: String,
  runKey: String,
  contextDocId: String
) async throws {
  // #region example
  let result = try await client.workflows.confirmApply(
    workflowKey: workflowKey,
    runKey: runKey,
    contextDocId: contextDocId
  )
  // result["confirmed"]
  // #endregion example
  _ = result
}
